client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,472 @@
+        a sheet name form input replacing the selected tab, so we hide the true tab bar */}
+      {(!isEditing || sheets.length === 1) && (
+        <Tabs
+          id={batch.name}
+          onChange={(newTabId: string) => {
+            setSelectedTabId(newTabId)
+          }}
+          selectedTabId={selectedTabId}
+        >
+          {tabs.map(tab => (
+            <Tab id={tab.id} key={tab.id}>
+              {tab.name}
+            </Tab>
+          ))}
+          <Tabs.Expander />
+          {sheets.length > 1 && (
+            <Button
+              icon="add"
+              disabled={areResultsFinalized}
+              minimal
+              onClick={addSheet}
+            >
+              Add Sheet
+            </Button>
+          )}
+        </Tabs>
+      )}
+
+      <BatchResultTallySheet
+        addSheet={addSheet}
+        areResultsFinalized={areResultsFinalized}
+        batch={batch}
+        contest={contest}
+        deleteSheet={deleteSheet}
+        isEditing={isEditing}
+        key={selectedTabId}
+        selectedTabId={selectedTabId}
+        setAreChangesUnsaved={setAreChangesUnsaved}
+        setIsEditing={setIsEditing}
+        sheets={sheets}
+        updateSheet={updateCurrentSheet}
+      />
+    </Detail>
+  )
+}
+
+interface IBatchResultTallySheetProps {
+  addSheet: () => Promise<void>
+  areResultsFinalized: boolean
+  batch: IBatch
+  contest: IContest
+  deleteSheet: () => Promise<void>
+  isEditing: boolean
+  selectedTabId: string
+  setAreChangesUnsaved: (areChangesUnsaved: boolean) => void
+  setIsEditing: (isEditing: boolean) => void
+  sheets: IBatchResultTallySheetWithId[]
+  updateSheet: (updatedSheet: IBatchResultTallySheet) => Promise<void>
+
+  // Require a key to ensure that the state within this component resets when a different sheet is
+  // selected
+  key: string // eslint-disable-line react/no-unused-prop-types
+}
+
+const BatchResultTallySheet: React.FC<IBatchResultTallySheetProps> = ({
+  addSheet,
+  areResultsFinalized,
+  batch,
+  contest,
+  deleteSheet,
+  isEditing,
+  selectedTabId,
+  setAreChangesUnsaved,
+  setIsEditing,
+  sheets,
+  updateSheet,
+}) => {
+  const tabs = tabsFromSheets(sheets)
+  const isTotalsSheet = selectedTabId === VOTE_TOTALS_TAB_ID
+  const selectedSheet = isTotalsSheet
+    ? undefined
+    : sheets.find(sheet => sheet.id === selectedTabId)
+
+  const formMethods = useForm<IBatchResultTallySheet>({
+    defaultValues: selectedSheet,
+  })
+  const { errors, formState, handleSubmit, register, reset } = formMethods
+  // Important gotcha! You have to access properties on the formState to subscribe to it:
+  // https://github.com/react-hook-form/react-hook-form/issues/9002
+  const { isSubmitting, isDirty } = formState
+
+  // Communicate up to the parent whether or not there are unsaved changes
+  useEffect(() => {
+    setAreChangesUnsaved(isDirty)
+  }, [isDirty, setAreChangesUnsaved])
```

Comment:
Is there any way to move this into an event handler? It's [not recommended](https://beta.reactjs.org/learn/you-might-not-need-an-effect#passing-data-to-the-parent) to use an effect for this. I know that might not be possible due to how react-hook-form works

Reply:
I couldn't think of a way to do so without bringing the react-hook-form form up a few levels. But for separate reasons (https://github.com/votingworks/arlo/pull/1661#discussion_r1011136794), I actually decided to stop using `formState.isDirty` altogether. I'm now just checking whether or not we're in edit mode: https://github.com/votingworks/arlo/pull/1661/commits/a814bee95694eab86dcb41bf9861ea37e2e0feb3

Reply:
Incredible sidestep! Can't believe we never thought of this option before. It actually feels better as a user to know that I can never accidentally exit edit mode. And the code is much simpler now 👏

Reply:
:tada: Feels so obvious in retrospect!